### PR TITLE
Loki: Preserve pipeline stages in context query

### LINKED ISSUE #69623: Loki: Log context should preserve pipe operations (e.g. line_format)
Following up on a new Loki log context UI where users can edit their query and preserve parser, users would like to preserve also pipe operations such as `line_format`.

### PR DESCRIPTION
**Pull request by svennergr**
**What is this feature?**

This PR adds a toggle to preserve other pipeline stages in the context query. With this users will keep the same `line_format` as they set in the original query.

**Special notes for your reviewer:**

Fixes #69623

https://github.com/grafana/grafana/assets/8092184/438e8494-f02c-4409-b3af-fed097b18a3f

### CODE REVIEW

File: public/app/plugins/datasource/loki/components/LokiContextUi.tsx [hunk 1]
```diff
@@ -345,6 +377,36 @@ export function LokiContextUi(props: LokiContextUiProps) {
...
+                  transparent={true}
+                  onChange={(e) => {
+                    reportInteraction('grafana_explore_logs_loki_log_context_pipeline_toggled', {
+                      logRowUid: row.uid,
+                      action: e.currentTarget.checked ? 'enable' : 'disable',
+                    });
+                    store.set(SHOULD_INCLUDE_PIPELINE_OPERATIONS, e.currentTarget.checked);
+                    setIncludePipelineOperations(e.currentTarget.checked);
+                    // need to retrigger a query to run by calling `updateFilter`
+                    updateFilter(contextFilters.filter(({ enabled }) => enabled));
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/components/LokiContextUi.tsx:404]: I think we should have a new function, something like `updateIncludePipelineOperations` that would trigger runContextQuery`.

**Review by ivanahuckova — COMMENTED** [review]
Nice work!! Left some comments bellow.

File: public/app/plugins/datasource/loki/components/LokiContextUi.tsx (hunk 1 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/components/LokiContextUi.tsx:404]: I am okay with leaving this this way with a comment, but my preference would be not manually triggering this trough context filters.

File: public/app/plugins/datasource/loki/components/LokiContextUi.tsx [hunk 2]
```diff
@@ -222,6 +261,7 @@ export function LokiContextUi(props: LokiContextUiProps) {
               });
               // We are removing the preserved labels from local storage so we can preselect the labels in the UI
               store.delete(LOKI_LOG_CONTEXT_PRESERVED_LABELS);
+              store.delete(SHOULD_INCLUDE_PIPELINE_OPERATIONS);
```
[the remark this answers is not in the document]
Comment (svennergr) [reply public/app/plugins/datasource/loki/components/LokiContextUi.tsx:264]: Fixed in 44b60547f8

File: public/app/plugins/datasource/loki/components/LokiContextUi.tsx (hunk 1 above)
Comment (svennergr) [reply public/app/plugins/datasource/loki/components/LokiContextUi.tsx]: I added the `runContextQuery` as a prop to `LokiContextUi` and just run that in 5a69c2ad4a2075b7e1137d952ae17826cf43fe0e

File: public/app/plugins/datasource/loki/LogContextProvider.ts [hunk 3]
```diff
@@ -196,7 +196,19 @@ export class LogContextProvider {
     });
   }
 
-  processContextFiltersToExpr = (row: LogRowModel, contextFilters: ContextFilter[], query: LokiQuery | undefined) => {
+  prepareExpression(
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LogContextProvider.ts:200]: 😍

File: public/app/plugins/datasource/loki/LogContextProvider.ts [hunk 4]
```diff
@@ -109,7 +117,12 @@ export class LogContextProvider {
     direction: LogRowContextQueryDirection,
     origQuery?: LokiQuery
   ): Promise<{ query: LokiQuery; range: TimeRange }> {
-    const expr = this.processContextFiltersToExpr(row, this.appliedContextFilters, origQuery);
+    let expr = this.prepareExpression(
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/LogContextProvider.ts:120]:
```suggestion
    const expr = this.prepareExpression(
```

**Review by ivanahuckova — APPROVED** [review]
LGTM

**Pull request merged**